src/MySqlConnector/MySqlClient/Caches/NormalizedName.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Text.RegularExpressions;
+
+namespace MySql.Data.MySqlClient.Caches
+{
+	internal class NormalizedName
+	{
+		private static readonly Regex NameRe = new Regex(@"^\s*(?:`([^`]+)`|([^\.]+)?)\s*\.?\s*(?:`([^`]+)`|([^\.]+)?)?\s*$",
```

Comment:
There are valid (but pathological) MySQL identifiers that will break this regex. 

> Identifier quote characters can be included within an identifier if you quote the identifier. If the character to be included within the identifier is the same as that used to quote the identifier itself, then you need to double the character.

http://dev.mysql.com/doc/refman/5.7/en/identifiers.html

May not be worth fixing, since it's such an edge case but FYI (in case you want to make a note or a known issue about it).